<?xml version="1.0" ?>
<robot name="indy7" xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:macro name="indy7_robot" params="robot_id:='indy7' description_pkg:='indy7_description' connected_to:='' xyz:='0 0 0' rpy:='0 0 0'">
    <xacro:unless value="${not connected_to}">
      <joint name="indy${robot_id}_joint_${connected_to}" type="fixed">
        <parent link="${connected_to}"/>
        <child link="indy${robot_id}_link0"/>
        <origin rpy="${rpy}" xyz="${xyz}"/>
      </joint>
    </xacro:unless>
	<xacro:property name="PI" value="3.1415926535897931"/>
	<xacro:property name="indy_material" value="metal_grey"/>
	<!-- <material name="grey"> <color rgba="0.5 0.5 0.5 1"/> </material> -->

	<link name="indy${robot_id}_link0">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_0.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="1.59306955"/>
			<inertia ixx="+0.00572623" iyy="+0.00558959" izz="+0.00966674" ixy="+0.00000251" iyz="-0.00000014" ixz="-0.00011380"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link1">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_1.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="0 0 0.072" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="11.8030102"/>
			<inertia ixx="+0.15418559" iyy="+0.12937017" izz="+0.05964415" ixy="-0.00000235" iyz="-0.04854267" ixz="+0.00001739"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link2">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_2.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="-0.225 0 0.09" rpy="${PI/2} 0 -${PI/2}"/>
			<geometry>
				<cylinder length="0.450" radius="0.073"/>
			</geometry>
		</collision>
		<collision>
			<origin xyz="0 0 -0.01" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.2" radius="0.085"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="7.99292141"/>
			<inertia ixx="+0.29356980" iyy="+0.28094142" izz="+0.03620609" ixy="-0.00000040" iyz="+0.03727972" ixz="+0.00001441"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link3">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_3.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="0 0 0.03" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.2" radius="0.065"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="2.99134127"/>
			<inertia ixx="+0.03424593" iyy="+0.03406024" izz="+0.00450477" ixy="+0.00000149" iyz="+0.00186009" ixz="+0.00000724"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link4">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_4.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="0 0 -0.092" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.35" radius="0.06"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="2.12317035"/>
			<inertia ixx="+0.00670405" iyy="+0.00279246" izz="+0.00619341" ixy="+0.00000375" iyz="-0.00127967" ixz="+0.00000150"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link5">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_5.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="-0 0 -0.0225" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.183" radius="0.055"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="2.28865091"/>
			<inertia ixx="+0.00994891" iyy="+0.00978189" izz="+0.00271492" ixy="+0.00000014" iyz="-0.00093546" ixz="+0.00000321"/>
		</inertial>
	</link>

	<link name="indy${robot_id}_link6">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://indy7_description/meshes/stl/Indy7_6.stl"/>
			</geometry>
			<material name="${indy_material}"/>
		</visual>
		<collision>
			<origin xyz="0 0 -0.038" rpy="0 0 0"/>
			<geometry>
				<cylinder length="0.25" radius="0.06"/>
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.40083918"/>
			<inertia ixx="0.00043534" iyy="0.00044549" izz="0.00059634" ixy="+0.00000013" iyz="+0.00000051" ixz="-0.00000002"/>
		</inertial>
	</link>
	<link name="indy${robot_id}_tcp"/>

	<joint name="indy${robot_id}_joint0" type="revolute">
		<parent link="indy${robot_id}_link0"/>
		<child link="indy${robot_id}_link1"/>
		<origin xyz="0 0 0.0775" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${175*PI/180}" upper="${175*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_joint1" type="revolute">
		<parent link="indy${robot_id}_link1"/>
		<child link="indy${robot_id}_link2"/>
		<origin xyz="0 -0.10900000 0.22200000" rpy="${PI/2} ${PI/2} 0"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${175*PI/180}" upper="${175*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_joint2" type="revolute">
		<parent link="indy${robot_id}_link2"/>
		<child link="indy${robot_id}_link3"/>
		<origin xyz="-0.45000000 +0.00000000 -0.03050000" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${175*PI/180}" upper="${175*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_joint3" type="revolute">
		<parent link="indy${robot_id}_link3"/>
		<child link="indy${robot_id}_link4"/>
		<origin xyz="-0.26700000 +0.00000000 -0.07500000" rpy="-${PI/2} 0 ${PI/2}"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${175*PI/180}" upper="${175*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_joint4" type="revolute">
		<parent link="indy${robot_id}_link4"/>
		<child link="indy${robot_id}_link5"/>
		<origin xyz="+0.00000000 -0.11400000 +0.08300000" rpy="${PI/2} ${PI/2} 0"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${175*PI/180}" upper="${175*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_joint5" type="revolute">
		<parent link="indy${robot_id}_link5"/>
		<child link="indy${robot_id}_link6"/>
		<origin xyz="-0.16800000 +0.00000000 +0.06900000" rpy="-${PI/2} 0 ${PI/2}"/>
		<axis xyz="0 0 1"/>
		<limit effort="10" velocity="1" lower="-${215*PI/180}" upper="${215*PI/180}"/>
	</joint>

	<joint name="indy${robot_id}_tcp" type="fixed">
		<parent link="indy${robot_id}_link6"/>
		<child link="indy${robot_id}_tcp"/>
		<origin xyz="0 0 0.06" rpy="0 0 0"/>
	</joint>

	<transmission name="indy${robot_id}_tran0">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint0">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor0">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<transmission name="indy${robot_id}_tran1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint1">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor1">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<transmission name="indy${robot_id}_tran2">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint2">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor2">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<transmission name="indy${robot_id}_tran3">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint3">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor3">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<transmission name="indy${robot_id}_tran4">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint4">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor4">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<transmission name="indy${robot_id}_tran5">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="indy${robot_id}_joint5">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="indy${robot_id}_motor5">
			<hardwareInterface>PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
  </xacro:macro>
</robot>
